Fix staticcheck failures for pkg/scheduler/...

praseodym opened the pull request:
**What type of PR is this?**

/kind cleanup
/priority important-longterm
/sig scheduling

**What this PR does / why we need it**:

Cleanup the `pkg/scheduler/...` issues identified by staticcheck in #81189

**Which issue(s) this PR fixes**:

#36858, #81657

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

praseodym committed:
Fix staticcheck failures for pkg/scheduler/...

draveness reviewed (commented):
/ok-to-test

Huang-Wei reviewed (commented):
@praseodym Thanks for the PR.

I'd like all staticcheck fixes related with scheduler to be included in one PR. Could you help to fix the following packages:

- cmd/kube-scheduler
- pkg/apis/scheduling/v1alpha1
- pkg/apis/scheduling/v1beta1
- pkg/controller/volume/scheduling
- test/e2e/scheduling
- test/integration/scheduler
- test/integration/scheduler_perf
- test/integration/volumescheduling

praseodym commented on pkg/scheduler/scheduler_test.go:
```diff
@@ -358,7 +358,7 @@ func TestSchedulerNoPhantomPodAfterExpire(t *testing.T) {
 			pods, err := scache.List(labels.Everything())
 			if err != nil {
-				t.Fatalf("cache.List failed: %v", err)
+				t.Errorf("cache.List failed: %v", err)
```
That's a lot nicer indeed, thanks! Applied and squashed into the first commit.

praseodym commented:
> I'd like all staticcheck fixes related with scheduler to be included in one PR. Could you help to fix the following packages: [..]

I've created a second commit that fixes everything but `test/integration/scheduler_perf`, which has a tricky failure:
```
test/integration/scheduler_perf/scheduler_bench_test.go:359:3: should not assign to b.N (SA3001)
```
The [b.N assignment is used to enforce the minimum number of pods](https://github.com/kubernetes/kubernetes/blob/d935b1054cece13bf32e1925b5f089b475c02f3d/test/integration/scheduler_perf/scheduler_bench_test.go#L358-L360) and as far as I can see there is no proper way to achieve to set this minimum value using the regular Go benchmark functions. Would you have any suggestions for this one?

Huang-Wei commented:
@praseodym Yeah, we're aware of that. You can leave it as is for now.

We will cover it in a scheduler performance improvement issue. (cc/ @draveness that staticcheck detects this issue)

praseodym commented:
@Huang-Wei Alright, thanks. Any preference as to leaving that package in `.staticcheck_failures` vs. removing it and adding an ignore comment in the code itself?

Huang-Wei commented:
> Any preference as to leaving that package in .staticcheck_failures vs. removing it and adding an ignore comment in the code itself?

Should be good to leave it in `.staticcheck_failures`.

Huang-Wei commented:
/lgtm
/approve

@msau42 could you help /approve the changes on package `pkg/controller/volume/scheduling/`? Thanks.

@tallclair, in terms of changes in /hack, /test and pkg/apis/, do we have dedicated approver reviewing for staticcheck, or I just randomly ping the approvers? Thanks.

msau42 commented on pkg/controller/volume/scheduling/scheduler_binder_test.go:
```diff
@@ -51,7 +51,6 @@ var (
 	preboundPVC         = makeTestPVC("prebound-pvc", "1G", "", pvcPrebound, "pv-node1a", "1", &waitClass)
 	preboundPVCNode1a   = makeTestPVC("unbound-pvc", "1G", "", pvcPrebound, "pv-node1a", "1", &waitClass)
 	boundPVC            = makeTestPVC("bound-pvc", "1G", "", pvcBound, "pv-bound", "1", &waitClass)
-	boundPVC2           = makeTestPVC("bound-pvc2", "1G", "", pvcBound, "pv-bound2", "1", &waitClass)
```
https://github.com/kubernetes/kubernetes/pull/81680 already covers this

Huang-Wei replied:
Got it.

@praseodym Can you revert the change in latest commit on pkg/controller/volume/scheduling (as well as the entry in .staticcheck_failures)

praseodym replied:
Done (and rebased)!

SataQiu reviewed (approved):
/approve
for hack

praseodym committed:
Fix staticcheck failures for scheduler packages

Errors from staticcheck:
cmd/kube-scheduler/app/server.go:297:27: prometheus.Handler is deprecated: Please note the issues described in the doc comment of InstrumentHandler. You might want to consider using promhttp.Handler instead.  (SA1019)
pkg/apis/scheduling/v1alpha1/defaults.go:27:6: func addDefaultingFuncs is unused (U1000)
pkg/apis/scheduling/v1beta1/defaults.go:27:6: func addDefaultingFuncs is unused (U1000)
test/e2e/scheduling/predicates.go:757:6: func verifyReplicasResult is unused (U1000)
test/e2e/scheduling/predicates.go:765:6: func getPodsByLabels is unused (U1000)
test/e2e/scheduling/predicates.go:772:6: func runAndKeepPodWithLabelAndGetNodeName is unused (U1000)
test/e2e/scheduling/limit_range.go:172:3: this value of pod is never used (SA4006)
test/e2e/scheduling/limit_range.go:177:3: this value of pod is never used (SA4006)
test/e2e/scheduling/limit_range.go:196:3: this value of pod is never used (SA4006)
test/e2e/scheduling/limit_range.go:201:3: this value of pod is never used (SA4006)
test/e2e/scheduling/limit_range.go:240:3: this value of pod is never used (SA4006)
test/e2e/scheduling/taints.go:428:13: this value of err is never used (SA4006)
test/e2e/scheduling/ubernetes_lite.go:219:2: this value of pods is never used (SA4006)
test/integration/scheduler/extender_test.go:78:4: this value of resp is never used (SA4006)
test/integration/volumescheduling/volume_binding_test.go:529:15: this result of append is never used, except maybe in other appends (SA4010)
test/integration/volumescheduling/volume_binding_test.go:538:15: this result of append is never used, except maybe in other appends (SA4010)

praseodym commented:
/assign @liggitt

praseodym commented:
Rebased to fix conflicts.

@liggitt could you approve/review for `pkg/apis` and `test`? Thanks :slightly_smiling_face:

liggitt commented on cmd/kube-scheduler/app/server.go:
```diff
@@ -293,7 +293,7 @@ func buildHandlerChain(handler http.Handler, authn authenticator.Request, authz
 func installMetricHandler(pathRecorderMux *mux.PathRecorderMux) {
 	configz.InstallHandler(pathRecorderMux)
-	defaultMetricsHandler := legacyregistry.Handler().ServeHTTP
+	defaultMetricsHandler := promhttp.Handler().ServeHTTP
```
@logicalhan does this change look right to you?

logicalhan replied:
No. This is going to break our metrics. We should just add the ignore lint directive above the line for the time being.

praseodym replied:
Done, added as:
```go
//lint:ignore SA1019 See the Metrics Stability Migration KEP
```

logicalhan commented:
/hold

liggitt commented on pkg/scheduler/api/v1/types.go:
```diff
@@ -145,7 +145,7 @@ type UtilizationShapePoint struct {
 type ResourceSpec struct {
 	// Name of the resource to be managed by RequestedToCapacityRatio function.
-	Name apiv1.ResourceName `json:"name,casttype=ResourceName"`
+	Name apiv1.ResourceName `json:"name" protbuf:"casttype=ResourceName"`
```
not sure what this change is about... the scheduler config API isn't proto-serialized (and "protbuf" is a typo anyway)

liggitt replied:
possibly just drop the casttype directive?

praseodym replied:
Good catch on the typo! Dropped the casttype directive entirely instead.

Huang-Wei replied:
@liggitt is right, we shoudn't have `protbuf`. Interestingly, when I first reviewed the PR, I saw only "casttype=xyz" was removed.

praseodym replied:
@Huang-Wei that’s because this second `casttype` instance wasn’t there in the initial version of this PR. It was introduced in https://github.com/kubernetes/kubernetes/pull/77688 which got in through a rebase, but I fixed it differently than the first `casttype` instance (my bad!).

logicalhan reviewed (commented):
/hold cancel
/lgtm from sig-instrumentation perspective.

liggitt commented:
/approve

liggitt commented:
/lgtm
